tests/backup/backup_sse_test.go

```diff
@@ -207,18 +209,22 @@ var _ = Describe("{CreateBackupAndRestoreForAllCombinationsOfSSES3AndDenyPolicy}
 				}
 				wg.Wait()
 			})
-			Step("Create restore with replace policy set to retain", func() {
-				log.InfoD("Create restore with replace policy set to retain")
-				restoreName := fmt.Sprintf("restore-with-replace-%s-%s-%v", RestoreNamePrefix, RandomString(randomStringLength), time.Now().Unix())
+			Step("Create and validate restore with replace policy set to retain", func() {
+				log.InfoD("Create and validate restore with replace policy set to retain")
+				restoreName := fmt.Sprintf("restore-with-replace-%s-%s-%v", RestoreNamePrefix, RandomString(randomStringLength), RandomString(5))
 				err = CreateRestoreWithReplacePolicy(restoreName, backupsWithSse[0], make(map[string]string), SourceClusterName, orgID, ctx, make(map[string]string), 2)
 				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore [%s]", restoreName))
 				restoreList = append(restoreList, restoreName)
+				appContextsToBackup = FilterAppContextsByNamespace(scheduledAppContexts, []string{bkpNamespaces[midpoint]})
```

Comment:
Is it not bkpNamespacesWithSse instead of bkpNamespaces?

As the backups are created in parallel, backupsWithSse[0] might not be associated with bkpNamespacesWithSse[0]. A safer approach would be to rely on FetchNamespacesFromBackup or to maintain a backupNamespaceMap.

Reply:
corrected